### PR TITLE
Ensuring schema is passed into the Provider constructor for GetSchema

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #8

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,6 +25,7 @@ ensure::
 # Provider
 
 build_provider:: ensure
+	cp ${WORKING_DIR}/schema.json provider/cmd/${PROVIDER}/
```
Comment (justinvp) [inline Makefile:28]: Nit:

```suggestion
	cp ${SCHEMA_PATH} provider/cmd/${PROVIDER}/
```

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,6 @@ ci-scripts/
 *.tar.gz
 
 /provider/cmd/pulumi-resource-*/schema.go
+# This is an artifact only so that we can pass the
+# schema for the provider GetSchema gRPC request
+/provider/cmd/pulumi-gen-*/schema.json
```
Comment (justinvp) [inline .gitignore:22]: I may very well be missing something, but shouldn't this line be the following?

```suggestion
/provider/cmd/pulumi-resource-*/schema.json
```

File: .gitignore [hunk 3]
```diff
@@ -17,3 +17,6 @@ ci-scripts/
 *.tar.gz
 
 /provider/cmd/pulumi-resource-*/schema.go
```
Comment (justinvp) [inline .gitignore:19]: Looks like I copy/pasted this file from the Go boilerplate. I don't think this line is relevant for this repo. Mind deleting it while making changes here?

**Review by justinvp — APPROVED** [review]
LGTM otherwise. (I haven't tested it.)

File: provider/cmd/pulumi-resource-xyz/provider.ts [hunk 4]
```diff
@@ -18,7 +18,7 @@ import * as provider from "@pulumi/pulumi/provider";
 import { StaticPage, StaticPageArgs } from "./staticPage";
 
 export class Provider implements provider.Provider {
-    constructor(readonly version: string) { }
+    constructor(readonly version: string, readonly schema: string) { }
```
Comment (lukehoban) [inline provider/cmd/pulumi-resource-xyz/provider.ts:21]: Totally curious - mostly for my own education - how does this work?  We don't seem to be passing `schema` to anything - what is it that ensures this causes `GetSchema` to return the schema?

File: .gitignore (hunk 3 above)
Comment (stack72) [reply .gitignore:19]: Done!

File: .gitignore (hunk 2 above)
Comment (stack72) [reply .gitignore:22]: good catch!

File: Makefile (hunk 1 above)
Comment (stack72) [reply Makefile:28]: Simplier :)

**Pull request merged**